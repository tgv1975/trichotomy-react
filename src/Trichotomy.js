import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import trichotomy from 'trichotomy/dist/trichotomy.min.css';

export default class Trichotomy extends PureComponent {
    static defaultProps = {
        tag: 'div',
    }

    static propTypes = {
        tag: PropTypes.string,
        direction: PropTypes.oneOf(['vertical', 'horizontal']).isRequired,
    }

    render() {
        const {
            children,
            className,
            tag,
            direction,
        } = this.props;

        const Tag = tag;

        let cssClass = (className && className.concat(' ', trichotomy.trichotomy))
            || trichotomy.trichotomy;

        switch (direction) {
            case 'horizontal':
                cssClass = `${cssClass} ${trichotomy.horizontal}`;
                break;
            default:
                cssClass = `${cssClass} ${trichotomy.vertical}`;
                break;
        }

        return (
            <Tag className={cssClass}>
                {children}
            </Tag>
        );
    }
}
