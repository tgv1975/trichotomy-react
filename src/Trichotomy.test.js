import React from 'react';
import Trichotomy from './Trichotomy';
import renderer from 'react-test-renderer';

describe('Trichotomy', () => {
    it('Trichotomy is truthy', () => {
        expect(Trichotomy).toBeTruthy();
    });

    test('Renders Trichotomy container <DIV> element (default)', () => {
        const component = renderer.create(
            <Trichotomy direction="vertical"></Trichotomy>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('div');
    });

    test('Renders Trichotomy container <NAV> element (custom)', () => {
        const component = renderer.create(
            <Trichotomy tag="nav" direction="vertical"></Trichotomy>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('nav');
    });

    test('Renders Trichotomy container element with "trichotomy vertical" CSS classes', () => {
        const component = renderer.create(
            <Trichotomy direction="vertical"></Trichotomy>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('div');
    });

    test('Renders Trichotomy container element with "trichotomy horizontal" CSS classes', () => {
        const component = renderer.create(
            <Trichotomy direction="horizontal"></Trichotomy>
        );

        let tree = component.toJSON();

        expect(tree).toMatchSnapshot();

        expect(tree.type).toEqual('div');
    });
});
