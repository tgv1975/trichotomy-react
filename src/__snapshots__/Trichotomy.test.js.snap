// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Trichotomy Renders Trichotomy container <DIV> element (default) 1`] = `
<div
  className="trichotomy vertical"
/>
`;

exports[`Trichotomy Renders Trichotomy container <NAV> element (custom) 1`] = `
<nav
  className="trichotomy vertical"
/>
`;

exports[`Trichotomy Renders Trichotomy container element with "trichotomy horizontal" CSS classes 1`] = `
<div
  className="trichotomy horizontal"
/>
`;

exports[`Trichotomy Renders Trichotomy container element with "trichotomy vertical" CSS classes 1`] = `
<div
  className="trichotomy vertical"
/>
`;
